src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs

```diff
@@ -82,14 +82,23 @@ public TypeSyntaxSimplifierWalker(CSharpSimplifyTypeNamesDiagnosticAnalyzer anal
             _cancellationToken = cancellationToken;
 
             var root = semanticModel.SyntaxTree.GetRoot(cancellationToken);
-            _aliasedNames = GetAliasedNames(root as CompilationUnitSyntax);
+            _aliasedNames = GetAliasedNames((CompilationUnitSyntax)root);
         }
 
-        private static ImmutableHashSet<string> GetAliasedNames(CompilationUnitSyntax? compilationUnit)
+        private ImmutableHashSet<string> GetAliasedNames(CompilationUnitSyntax compilationUnit)
         {
             var aliasedNames = s_emptyAliasedNames;
-            if (compilationUnit is null)
-                return aliasedNames;
+
+            var scopes = _semanticModel.GetImportScopes(0, _cancellationToken);
+            foreach (var scope in scopes)
+            {
+                foreach (var alias in scope.Aliases)
+                {
+                    var name = alias.Target.Name;
+                    if (!string.IsNullOrEmpty(name))
+                        aliasedNames = aliasedNames.Add(name);
+                }
+            }
 
             foreach (var usingDirective in compilationUnit.Usings)
```

Comment:
Are these not already included because of the above? or does position 0 in GetImportScopes mean it only returns the global usings?

Reply:
yeah, this will pull in the other global usings, it doesn't pull in the usings for the using block we're in.